html/infrastructure/common-dom-interfaces/collections/htmlallcollection.html

```diff
@@ -181,9 +207,36 @@
 }, "legacy caller with invalid \"array index property name\"");
 
 test(function() {
+  assert_equals(document.all(undefined), divs[3]);
+}, "legacy caller with undefined");
+
+test(function() {
+  assert_equals(document.all(null), divs[4]);
+}, "legacy caller with null");
+
+test(function() {
   assert_equals(document.all(), null);
 }, "legacy caller with no argument");
 
+test(function() {
+  if (typeof Proxy !== "undefined") {
```

Comment:
This should be removed I think. Otherwise you can pass the test by not implementing proxies.

Reply:
There's still the more straightforward test below that's not guarded by this `if`.

https://github.com/w3c/web-platform-tests/blob/78f5b68c8159af473625223eccd7488075fce5b7/html/infrastructure/common-dom-interfaces/collections/htmlallcollection.html#L231-L233

Reply:
Sure, but why would we guard this one?

Reply:
Yeah, guards in tests skew results and create weird incentives. Let's not go there?